custom-addons/esb/sale/models/sale.py

```diff
@@ -24,6 +23,10 @@ class SaleOrder(models.Model):
         string="Bank Account",
         domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]",
     )
+    type_id = fields.Many2one(
+        comodel_name="sale.order.type",
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]",
+    )
```

Comment:
If you create a pull request at OCA/sale-workflow to fix it, this code is unnecessary. You should edit `repos.yaml` instead.